retail/retail-java-applications/terraform/test/pubsub_test.go

```diff
@@ -0,0 +1,90 @@
+/**
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package test
+
+import (
+	"cloud.google.com/go/pubsub"
+	"context"
+	"fmt"
+	"github.com/gruntwork-io/terratest/modules/gcp"
+	"github.com/gruntwork-io/terratest/modules/random"
+	"github.com/gruntwork-io/terratest/modules/terraform"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func TestPubsub(t *testing.T) {
```

Comment:
This test is doing too many things. Refactor this into separate tests, one for each topic/subscription pair. For test code, DAMP is generally preferred over DRY. See: https://stackoverflow.com/questions/6453235/what-does-damp-not-dry-mean-when-talking-about-unit-tests. 

Also see https://github.com/GoogleCloudPlatform/golang-samples/blob/master/pubsub/subscriptions/subscription_test.go for an example of the common setup -> multiple tests pattern

Also read about the (generalization) principle of one assert per test as a heuristic: https://freecontent.manning.com/making-better-unit-tests-part-1-the-aaa-pattern/